### PR TITLE
feat: Complete test accounts (SC-3610)

### PR DESCRIPTION
**Pull request by deluca-mike**
- removed interfaces from interfaces
- StakeLockerOwner is better named as Pool
- StakeLockerPoolDelegate is better named as PoolDelegate
- StakeLockerStaker is better named as Staker

# Description

# Integrations Checklist

- [ ] Have any function signatures changed? If yes, outline below.
- [ ] Have any features changed or been added? If yes, outline below.
- [ ] Have any events changed or been added? If yes, outline below.
- [ ] Has all documentation been updated?

# Changelog

### CODE REVIEW

File: contracts/StakeLocker.sol [hunk 1]
```diff
@@ -18,10 +18,9 @@ contract StakeLocker is IStakeLocker, StakeLockerFDT, Pausable {
 
     uint256 constant WAD = 10 ** 18;  // Scaling factor for synthetic float division.
 
-    IERC20  public override immutable stakeAsset;
-
     address public override immutable liquidityAsset;
     address public override immutable pool;
+    address public override immutable stakeAsset;
```
Comment (lucas-manuel) [inline contracts/StakeLocker.sol]: Can we just verify if this reordering changes the bytecode?

File: contracts/test/mocks/Mocks.sol [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    constructor(address _globals) public {
+        globals = _globals;
+    }
+
+}
+
+contract PoolMock {
+
+    bool    public constant  isPoolFinalized = false;
```
Comment (lucas-manuel) [inline contracts/test/mocks/Mocks.sol]: `bool public constant isPoolFinalized = false;`

File: contracts/test/StakeLocker.t.sol [hunk 3]
```diff
@@ -71,18 +29,18 @@ contract StakeLockerConstructorTest is DSTest {
...
-    StakeLocker             internal locker;
-    StakeLockerStaker       internal staker;
+    PoolDelegate     internal delegate;
+    PoolFactoryMock  internal factory;
+    PoolMock         internal pool;
+    MapleGlobalsMock internal globals;
+    MockToken        internal stakeToken;
+    MockToken        internal liquidityToken;
+    StakeLocker      internal locker;
+    Staker           internal staker;
```
Comment (lucas-manuel) [inline contracts/test/StakeLocker.t.sol]: Alphabetical

**Review by lucas-manuel — CHANGES_REQUESTED** [review]

File: contracts/StakeLocker.sol (hunk 1 above)
Comment (deluca-mike) [reply contracts/StakeLocker.sol]: Will check.

**Review by lucas-manuel — DISMISSED** [review]

File: contracts/StakeLocker.sol (hunk 1 above)
Comment (deluca-mike) [reply contracts/StakeLocker.sol]: Interestingly, but in hindsight, obviously, the bytecode for the `StakeLocker` did not change, but the bytecode for the `StakeLockerFactory` did. So I revert this change.

**Pull request merged**